src/Cli/dotnet/commands/dotnet-workload/clean/WorkloadCleanCommand.cs

```diff
@@ -0,0 +1,132 @@
+
+            _sdkVersion = WorkloadOptionsExtensions.GetValidatedSdkVersion(parseResult.GetValue(WorkloadUninstallCommandParser.VersionOption), version, _dotnetPath, userProfileDir, true);
+            var sdkFeatureBand = new SdkFeatureBand(_sdkVersion);
+
+            var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(_dotnetPath, _sdkVersion.ToString(), _userProfileDir);
+            _workloadResolver = workloadResolver ?? WorkloadResolver.Create(workloadManifestProvider, _dotnetPath, _sdkVersion.ToString(), _userProfileDir);
+            _workloadInstaller = WorkloadInstallerFactory.GetWorkloadInstaller(Reporter, sdkFeatureBand, _workloadResolver, Verbosity, _userProfileDir, VerifySignatures, PackageDownloader, _dotnetPath);
+            
+        }
+
+        public override int Execute()
+        {
+            ExecuteGarbageCollection();
+            return 0;
+        }
+
+        private void ExecuteGarbageCollection()
+        {
+            if (_cleanAll)
+            {
+                _workloadInstaller.GarbageCollectInstalledWorkloadPacks(cleanAllPacks: true);
+            }
+            else
+            {
+                _workloadInstaller.GarbageCollectInstalledWorkloadPacks();
+            }
+
+            DisplayUninstallableVSWorkloads();
+        }
+
+        /// <summary>
+        /// Print VS Workloads with the same machine arch which can't be uninstalled through the SDK CLI to increase user awareness that they must uninstall via VS.
+        /// </summary>
+        private void DisplayUninstallableVSWorkloads()
+        {
+#if !DOT_NET_BUILD_FROM_SOURCE
+            if (OperatingSystem.IsWindows())
+            {
+                // All VS Workloads should have a corresponding MSI based SDK. This means we can pull all of the VS SDK feature bands using MSI/VS related registry keys.
+                var installedSDKVersionsWithPotentialVSRecords = MsiInstallerBase.GetInstalledSdkVersions();
+                HashSet<string> vsWorkloadUninstallWarnings = new();
+
+                foreach (string sdkVersion in installedSDKVersionsWithPotentialVSRecords)
+                {
+                    try
+                    {
+#pragma warning disable CS8604 // We error in the constructor if the dotnet path is null.
+                        // The below environment expansion should be architecture agnostic for the C:\Program Files path, but only works on windows 7+. https://learn.microsoft.com/en-us/windows/win32/winprog64/wow64-implementation-details?redirectedfrom=MSDN
+                        // In addition, we don't need to worry about when dotnet will be dotnet/x64 (x64 on processor arch of arm64) because MSI install thru VS will only support installs on processor arch, not chip/emulation arch.
+                        string defaultDotnetWinPath = Path.Combine(Environment.ExpandEnvironmentVariables("%ProgramW6432%"), "dotnet");
```

Comment:
I'd suggest using Environment.SpecialFolder instead of the environment variables. Users can always override these.

Take a look at [this](https://github.com/dotnet/sdk/blob/033003f9fdbd09e33b3b2882c21d338d26bba88e/src/Cli/dotnet/commands/dotnet-workload/install/MsiInstallerBase.cs#L202) for an example

Reply:
Thanks, I think it would be even better to just call `GetDotNetHome`